src/com/redhat/oasis/MultistreamMoleculePipeline.groovy

```diff
@@ -287,6 +294,22 @@ class MultistreamMoleculePipeline implements Serializable {
         return config.preTestHook(config)
     }
 
+    def preScenarioHook(scenario):
+        if (config.preScenarioHook.maximumNumberOfParameters == 2) {
+            // short out if the hook is hookDefault. In this case, the two args
+            // are "(config, ...args)".
+            config.preScenarioHook(config)
+        } // else { assume 4-arg hook spec hereafter }
+
+        // wrap the molecule method in a closure to give hook callers the ability
+        // to call the molecule method for the current scenario.
+        def molecule_partial = {args_str ->
+            molecule(scenario, args_str)
+        }
+
+        config.preScenarioHook(config, scenario, molecule_partial, job)
```

Comment:
job should not be passed to hooks; hooks are defined in the job namespace and don't need to do any gymnastics to get at it.

...I swear I reviewed this in `git diff` before I pushed. :)